### DISCUSSION TITLE
Hello!

### DISCUSSION THREAD

**Discussion by printingotb** [5axes/UniversalCuraSettings#4] [General]
I take it that you are Laurent then?

How does this work? I'm not familiar with github other than submitting bug reports.
I have a relatively finished Universal settings profile now, do I upload it here? It's better than my old profiles.

Regards,
Jonas

**Comment by 5axes** [comment]
Yes I 'am.  Laurent is my real Name (5axes related to one of my previous Job and when i needed to use Github this account was already existing so I keep it without changing the nickname).  

This Idea of a tool to set the parameter not dependant of the profile was present for a long time in my mind but I never find the time or the good idea to change this idea into something serious. The discussion last week on facebook allowed me to start some work  on it. For the moment it's just something to test the concept and see if I can make it real.  

The general idea is to use a plugin that would modify the values rather than using a profile dependent on a machine configuration. By using a plugin you are then free to code all the conditions you want.

Now I don't know if the result will be too complicated in the end. 

For the moment the plugin uses an interface where the user has the choice between 3 criteria:

- The type of extruder (Direct drive or Bowden) 
- The material (I tested the solution to get the material from the Cura parameters but in the end it was too complicated especially in the case of multi extruder machine)
- And the setting mode (one I called mechanical and the other figurine, according to the profiles you posted on the facebook group)

![image](https://user-images.githubusercontent.com/11015345/112033438-b3c17c80-8b3d-11eb-936b-1a2818955d0b.png)

**Reply by printingotb** [reply]
5axes as in 5-axis CNC machining?

Now this is interesting, just let me know what you need of me. I don't code but I understand the basics to some extent. 

My original idea with my universal settings profile is to create one profile with the best universal tweaks that I can come up with and share it with a few instructions about temp, flow, retractions and speed. All settings should have defaults that is safe with the vast majority of printers. (Speed settings should not cause layer shifts on cheap printers etc..)

Here is my current latest version (arachne):
[JonasUniversalCuraSettings0.1.zip](https://github.com/5axes/JonasUniversalCuraSettings/files/6184647/JonasUniversalCuraSettings0.1.zip)


This may be beyond the concept stage but:
Is the idea that the Settings mode provides something like a normal settings profile where the user can still change everything manually? Or is the idea that the user can only change a select set of settings and the Settings mode runs the rest in the background? 

One question to think of is how many different settings modes there should be. 
For example:
Minis (fine details and good cooling)
Decorative (nice looking and relatively fast)
Mechanical (dimensionally accurate, stiff and durable)
Prototype (fast and ugly)
Vases (spiralize outer contour)
One could add more but less is better?

Another idea is to add additional parameters like for example:
Overhangs: None /  Easy / Difficult
Forces/load: None / Low / High
But again, maybe less is better.

**Reply by 5axes** [reply]
_Is the idea that the Settings mode provides something like a normal settings profile where the user can still change everything manually? Or is the idea that the user can only change a select set of settings and the Settings mode runs the rest in the background?_

The plugin is just ajusting the parameters according to your default value. But we can modify the values as we want according to the code.

**Comment by 5axes** [comment]
Yes uploaded your profile could be now the good timing, I have almost finished what I wanted to do in the main code and now I'm working on the parameter to modifiy by itself ..

**Reply by printingotb** [reply]
Attached it above, or do you want one for 4.8?

**Reply by 5axes** [reply]
First beta package : [https://github.com/5axes/JonasUniversalCuraSettings/releases/tag/V0.0.1](https://github.com/5axes/JonasUniversalCuraSettings/releases/tag/V0.0.1)

Just drag and drop the curapackage in your cura windows , and restart cura.

**Reply by printingotb** [reply]
This is cool :smiley: 

Noticed that window size and text colour needs a fix
![Skärmklipp](https://user-images.githubusercontent.com/32172100/112121011-0b53fc80-8bbf-11eb-9199-8d115338bee0.PNG)

**Reply by 5axes** [reply]
Yes strange for the text color 🤔

I think , I should also add a new combobox for the choice of the nozzle size

**Reply by printingotb** [reply]
I was just thinking that. I like to increase line width above the nozzle size to save printing time. Except for outer wall if there are fine details.
With the nozzle size known, the line width can be increased by say *1.25 for select line types depending on the settings mode. 
This can be an issue if the hotend doesn't have enough melt capacity though. Maybe we should add hotend type: PTFE lined, all metal hotend or high flow hotend? It would give us some base values to decide if and how much the line width can be increased in relation to layer height and speed. I also increase temp when increasing the volumetric flow, for a PTFE lined hotend, this should have a safe limit.

**Comment by 5axes** [comment]
Next step : Integrate the Nozlle Size as plugin parameter.

Start to think about how to modify the layer height in the plugin according to the nozzle size 
Start to think how to modify the material temperature according to the Nozzle size ( Add 5° every 0.2 mm ? ) 
Ie : 

Nozzle    Temp
0.2     200
0.4     205
0.6     210
0.8     215
1.0     220

**Reply by printingotb** [reply]
We wrote at the same time, does my above comment answer the question?
Layer height should perhaps be half the nozzle diameter as a base value?

**Reply by printingotb** [reply]
And temp in relation to volumetric flow rather than nozzle size. And a little extra if the line width is much larger than the nozzle diameter.

**Reply by printingotb** [reply]
And then there is the choice of nozzle material/heat conductivity.
Given that material temp can vary with brand, perhaps the user should choose a base temp. That would mean that nozzle material isn't important to know. The user probably knows what temp is a good default.

In my experience, people often print too cold though.

**Comment by 5axes** [comment]
![image](https://user-images.githubusercontent.com/11015345/112132606-17de5200-8bcb-11eb-9f95-9ee88d44a815.png)

**Reply by 5axes** [reply]
For the next version , I will not create a curapackage every time. If you want to test the "prototype" you should directly copy the files from the Github into the plugin directory.  ( normaly just the *.py and *.qml file are required )

**Reply by printingotb** [reply]
Sounds easy enough.

**Comment by 5axes** [comment]
That's why I always like to make a real prototype rather than just theorize about how it works. While writing the code, the questions come naturally. For example, is it normal to be able to use a 0.8 nozzle to make miniatures? If so, should I define parameters for this case or should I just say that it is not recommended?   So many questions come up once you try to put all this into a real solution.

**Comment by printingotb** [comment]
It looks like it renamed my universal settings profile. 
![Skärmklipp](https://user-images.githubusercontent.com/32172100/112146932-6c3dfd80-8bdc-11eb-9f02-f83762a0e29b.PNG)

**Reply by 5axes** [reply]
Yes, I need to change this behaviour , it's not a bug but it's not a nice solution.   ( It rename the current profile and not create a new one ... )

**Reply by 5axes** [reply]
I recommand if you want to make some test , don't use this plugin on your regular Cura Release, But if you have an older release like 4.5, 4.6 , 4.7 it's always better to use in this early devellopement stage to work on a dedicade test version.

**Comment by 5axes** [comment]
The result with the profile JonasUniversalCuraSettings0.1  was not to the quality expected on my machine ( Ender3) . So going back to a basic concept "Universal Cura Settings"

**Reply by printingotb** [reply]
What happened to the quality? Can it be related to max resolution and deviation?

**Reply by 5axes** [reply]
Not sure of the cause. At first I thought the problem was acceleration and jerking. I've brought it back up to my standards which are lower. But the result is still unsatisfactory.
Max resolution and deviation could be the origine :

        modified_count += self._setValue("meshfix_maximum_deviation",0.04)
        modified_count += self._setValue("meshfix_maximum_resolution",0.4)

I will try with my own values : 


        modified_count += self._setValue("meshfix_maximum_deviation",0.02)
        modified_count += self._setValue("meshfix_maximum_resolution",0.2)

**Reply by 5axes** [reply]
Well it's not just the deviation and the resolution : 

![Settings](https://user-images.githubusercontent.com/11015345/114429862-1c8c9980-9bbe-11eb-930f-659ac221514c.JPG)

- Left Basic Cura settings Standard Quality 0.2 
- Right with the current settings of the plugin

**Reply by printingotb** [reply]
Have you tried with a custom fff machine? The Creality machines had a lot of issues with low values on max dev and resolution in 4.6 and 4.7. I don't remember what the safe values are.

The ender only has one Z-motor correct? Perhaps Z-hop is causing this if the right side of the X-gantry isn't moving smoothly. There is typically a little lag on the side without a motor. If so, the issue should be worse if the model is placed on the far right side of the bed.

How does the seam look? Sometimes if retraction settings aren't perfect, lines similar to those can form and take a relatively long distance to even out.

Also, white filament's tend to print worse than other colours due to titanium dioxide being used as a colourant.

Perhaps you know all this already?

**Reply by 5axes** [reply]
Yes Z hop will be my next test. I don't want to start from the generic printer configuration. The main idea with this plugin was the possibility to set some parameters without using a profile dependant from a machine/quality. Something similar to the answer you can give on a forum : for PLA and mechanical part I recommand temp 205,  bed temp 55°C etc etc. Not sure now to reach this goal

**Reply by printingotb** [reply]
I don't have an Ender to test with, but if Z-hop is the culprit, we could leave that off by default. 

The risk is if other machines has issues with other settings, we may have trouble finding settings that works for everyone. And then the profile may not be as good anymore.

**Reply by 5axes** [reply]
My goal is not to provide the best Profile, but to set some settings to solve some issues or reach some Quality Goal. As example we can use the case "bed adhesion" , the idea is to set only the parameters we have to modifiy to increase or solved the issue.  So in this case I don't want to offer a global profile like we can have with the actual solution but ajust some settings . Ie : 

 Increase the size and switch to an adhesion type to brim
Universaladhesion_type  set to 'brim'
Universalbrim_line_count  set to 15

Increase the reduction on small feature
Universalsmall_feature_speed_factor_0  set to 30
Universalsmall_hole_max_size  set to 6.0
          
Increase the first Layer line width		  
Universalinitial_layer_line_width_factor  set to 105

Reduce the speed on the first layer 
Universalspeed_layer_0  set to 18

Reduce the Jerk and the acceleration on the first layer
Universaljerk_layer_0  set to 5
Universaljerk_print_layer_0  set to 5            
Universalacceleration_print_layer_0  set to 300
Universalacceleration_wall_0  set to 300

And certainly more....

Let's say it's an innovative way of defining the settings by focussing the parameter on a issue or a result and not just think global settings.   And it will be possible for example to use the "Mechanical" settings +  "Bed adhesion"  if you have this issue

**Reply by 5axes** [reply]
Z hop was the origin of the bad result.  It's much more better now . But still have some minor issues.

**Reply by printingotb** [reply]
Perhaps Z-hop works better with slower Z-hop speed? But that will be an issue with higher Z-hop height.
If combing is turned on we should get less Z-hops and less issues. Not in skin may also work.

I compared the printer settings in Ender3 and Custom fff. Can't find anything in there that can explain why Z-hop wont' work well with the printer, Max Z speed/acceleration  is 10/100 in the ender while they are the speed of light in the custom fff. Both use a Default Z-jerk of 0.4mm/s. Nothing in the start gcode either. Perhaps the design of the ender with only one motor is the issue. Or it's something in the Ender firmware.

What model is that?

**Reply by 5axes** [reply]
Again my goal is not to define the best settings, I think it's just impossible but to define basic recommended options like you can do by giving some advice on a facebook post. So Z hop are in my settings only defined if you have bed adhesion issue to avoid the risk of   part detachement

**Reply by printingotb** [reply]
Yes, just trying to find problematic settings so you can decide what settings the plugin may want to avoid if possible. 
I like Z-hop, it's useful for overhangs that curls up and to prevent supports from being knocked down as well. It also prevents blobs forming on the walls of the parts when the nozzle oozes during travels, by passing a small distance over the wall.

**Reply by 5axes** [reply]
skin_overlap => 5 %  seems to low switch back to 10%

**Comment by 5axes** [comment]
Extruder type : **unknow** -> no modification of the retract parameter
Nozzle Size : **Not specified** -> no modification of the nozzle diameter
Material : **unknow** -> no modification of the temperature parameter

![image](https://user-images.githubusercontent.com/11015345/140521200-fee3d853-0e94-4d71-b163-6c39d1e79cda.png)
